spec/controllers/api/v1/users_controller_spec.rb

```diff
@@ -54,38 +106,88 @@ def create_call(user_params)
       post :create, params: { user: user_params }
     end
 
-    context 'when is successful' do
+    context 'when the user is admin and was authenticated' do
       let(:new_user) { { email: 'test@test.org', password: '123456' } }
 
       it 'returns successful' do
+        authenticate_user(admin)
         create_call(new_user)
         expect(response).to have_http_status(:created)
       end
 
       it 'returns the user' do
+        authenticate_user(admin)
         create_call(new_user)
         expect(JSON.parse(response.body)['email']).to eql(new_user[:email])
       end
     end
 
-    context 'when is not successful' do
+    context 'when the user is not admin' do
+      let(:new_user) { { email: 'test@test.org', password: '123456' } }
+
+      it 'returns unauthorized' do
+        authenticate_user(user)
+        create_call(new_user)
+        expect(response).to have_http_status(:unauthorized)
+      end
+
+      it 'returns the error code 4011' do
+        authenticate_user(user)
+        create_call(new_user)
+        expect(JSON.parse(response.body)['error_code']).to be(4011)
+      end
+
+      it 'returns the error message' do
+        authenticate_user(user)
+        create_call(new_user)
+        error_message = 'User cannot perform this action due to being unauthorized'
+        expect(JSON.parse(response.body)['description']).to eql(error_message)
+      end
+    end
+
+    context 'when params are invalid' do
       let(:new_user) { { email: 'test.org', password: '123456' } }
 
       it 'returns the error code 4022' do
+        authenticate_user(admin)
         create_call(new_user)
         expect(JSON.parse(response.body)['error_code']).to be(4022)
       end
 
       it 'returns code 422' do
+        authenticate_user(admin)
         create_call(new_user)
```

Comment:
you could define this as a setup method. Also check if admin is not becoming a mistery guest (method is way too up on the file doesn't it?)